js/ui/drop_down_box.js

```diff
@@ -20,6 +22,41 @@ var DROP_DOWN_BOX_CLASS = "dx-dropdownbox",
  * @export default
  */
 var DropDownBox = DropDownEditor.inherit({
+    _supportedKeys: function() {
+        return extend({}, this.callBase(), {
+            tab: function(e) {
+                if(!this.option("opened")) {
+                    return;
+                }
+
+                var $focusableElement = this._getTabbable(e.shiftKey);
+
+                $focusableElement && $focusableElement.focus();
+                e.preventDefault();
+            }
+        });
+    },
+
+    _getTabbable: function(getLast) {
+        var $elements = this._getElements(),
+            length = $elements.length,
+            tabbable;
+
+        for(var i = 0; i < length; i++) {
+            var $element = getLast ? $elements[length - i - 1] : $elements[i];
+
+            if(!tabbable && selectors.tabbable(null, $element)) {
+                tabbable = $element;
+                break;
+            }
+        }
+
+        return tabbable;
+    },
```

Comment:
Also, I think that _getTabbable method should be replaced with the _getTabbableElements method. And first/last methods may be called in several places where it is needed